### DISCUSSION TITLE
Error: No QueryClient set, use QueryClientProvider to set one

### DISCUSSION THREAD

**Discussion by Deivaras** [TanStack/query#1975] [Q&A] [answered]
react-query@3.12.1 and @3.12.2 getting Error: No QueryClient set, use QueryClientProvider to set one

Everything works fine with 3.12.0 and lower.
My (Next@latest) _app.js file:
`
import { QueryClient, QueryClientProvider } from 'react-query';
import { Hydrate } from 'react-query/hydration';
import { ReactQueryDevtools } from 'react-query/devtools';

export default function MyApp({ Component, pageProps }) {
  const queryClientRef = React.useRef();

  if (!queryClientRef.current) {
    queryClientRef.current = new QueryClient();
  }

  return (
    <React.Fragment>
      <QueryClientProvider client={queryClientRef.current}>
        <Hydrate state={pageProps.dehydratedState}>
            <Component {...pageProps} />
        </Hydrate>
        <ReactQueryDevtools />
      </QueryClientProvider>
    </React.Fragment>
  );
}
`

**Comment by TkDodo** [comment, marked as answer]
Is it this devtools issue?
https://github.com/tannerlinsley/react-query/issues/1936

**Reply by Deivaras** [reply]
> Is it this devtools issue?
> #1936

Not really. You can check in here:
https://codesandbox.io/s/distracted-rgb-emipc?file=/pages/_app.js

**Reply by TkDodo** [reply]
interestingly, I just clicked `Fork` on that sandbox and it works.. what could that be?

https://codesandbox.io/s/confident-resonance-546my?file=/pages/about.js

![Screenshot 2021-03-17 at 22 10 05](https://user-images.githubusercontent.com/1021430/111538940-97a09280-876d-11eb-9685-353ec123c537.png)

**Reply by Deivaras** [reply]
> interestingly, I just clicked `Fork` on that sandbox and it works.. what could that be?

Honestly no idea! Yesterday it was not working. Now it's all good on all @3.12.1, 2 and 3 :
https://codesandbox.io/s/distracted-rgb-emipc?file=/pages/about.js

Tried my app got this:
![image](https://user-images.githubusercontent.com/44872100/111608074-be43e500-87d0-11eb-9bf1-967036c0f504.png)
Going back to @3.12.0 solves the issue.

**Reply by TkDodo** [reply]
it still looks like the devtools issue. in 3.12.1 we changed how context is consumed

**Reply by Deivaras** [reply]
> it still looks like the devtools issue. in 3.12.1 we changed how context is consumed

Still getting same error after adding:
`<QueryClientProvider client={queryClientRef.current} contextSharing>`
Everything works in here perfect tho!:
https://codesandbox.io/s/distracted-rgb-emipc?file=/pages/about.js

**Reply by Deivaras** [reply]
Ok, so what happened was this (in next.config.js):
`module.exports = {
  webpack: (config) => {
    config.resolve = {
      ...config.resolve,
      alias: {
        ...config.resolve.alias,
        react$: resolveFrom(path.resolve('node_modules'), 'react'),
        // REMOVED THIS:
        **// 'react-query$': resolveFrom(
        //   path.resolve('node_modules'),
        //   'react-query'
        // ),**
        'react-dom$': resolveFrom(path.resolve('node_modules'), 'react-dom')
      }
    };
    return config;
  },
  images: {
    domains: ['res.cloudinary.com', 'cdn.sanity.io']
    // deviceSizes: [640, 750, 828, 1080, 1200, 1920, 2048, 3840]
  }
};`
Rookie mistake ;)
